src/registry/LocalRegistry.ts

```diff
@@ -89,6 +91,65 @@ export class LocalRegistry extends BaseRegistry implements IRegistry {
     return addresses[chainName];
   }
 
+  addChains(
+    chains: Array<{ chainName: ChainName; metadata?: ChainMetadata; addresses?: ChainAddresses }>,
+  ): void {
+    const currentChains = this.listRegistryContent().chains;
+    for (const chain of chains) {
+      if (currentChains[chain.chainName])
+        throw new Error(`Chain ${chain.chainName} already exists in registry`);
+    }
+
+    this.updateChains(chains);
+  }
+
+  updateChains(
+    chains: Array<{ chainName: ChainName; metadata?: ChainMetadata; addresses?: ChainAddresses }>,
+  ): void {
+    for (const chain of chains) {
+      if (!chain.metadata && !chain.addresses)
+        throw new Error(
+          `Chain ${chain.chainName} must have metadata or addresses, preferably both`,
+        );
+    }
+
+    // Ensure caches are initialized and filled before making changes
+    this.listRegistryContent();
+    this.getMetadata();
+    this.getAddresses();
+
+    for (const chain of chains) {
+      const chainName = chain.chainName;
+      if (chain.metadata) {
+        this.createChainFile(
+          chainName,
+          'metadata',
+          chain.metadata,
+          this.metadataCache!,
```

Comment:
slightly dont like this block, could we have the cache be returned in a var or have createChainFile read from the cache?

Reply:
Reworked it a bit, thanks for the suggestion